packages/extension-core/src/libs/GeneralReport.ts

```diff
@@ -59,6 +68,31 @@ async function getGeneralReport() {
   const watchedAccounts = accounts.filter(({ meta }) => meta.origin === AccountType.Watched)
   const watchedAccountsCount = watchedAccounts.length
 
+  const poolStatus = await Promise.race([
+    balancePool.status,
+    new Promise<"initialising">((resolve) => setTimeout(() => resolve("initialising"), 2_000)),
+  ])
+  // NOTE: There are two not-ready states of the balances pool, which are both called `initialising`.
+  //   1. The pool is still getting chainata ready so that it can set up balances subscriptions.
+  //   2. The pool is ready and has subscriptions, but some balances are still being refreshed from the chains.
+  //
+  // if balancePool isn't initialised (chaindata loaded from db), balances stats may be incorrect (initialising state 1)
+  if (balancePool.hasInitialised.isPending()) return
+  // if balancePool isn't initialised (all balance queries set to `live`), balances stats may be incorrect (initialising state 2)
+  if (poolStatus === "initialising") return
+
+  // if the balancePool has cached balances, we probably have an incomplete state of the user's balances
+  // (i.e. we're still fetching them after a browser restart)
+  //
+  // 30 seconds after the pool is initialised, all of the cached balances are transformed into stale balances
+  //
+  // in the future, we should expose an observable on the pool which can indicate whether or not the pool is hydrated,
```

Comment:
I think this was done in Kheops' recent work?

Reply:
I think Kheops' new `isRestartPending` state covers this edge-case for code which subscribes to balances, because creating a subscription sets `isRestartPending` to `true`.  
But in the case of analytics here, we're not telling the pool to subscribe to balances for us, we're just asking for the pool's current state and hoping someone else has set up the subscription.

As far as I can tell, this means we might ask the pool for its current state when neither of these two are true:
- `isRestartPending`
- `initialising.length`

Reply:
So I think we still need this check here to avoid the edge-case where we try and do a general report on a pool which hasn't been subscribed to since the last background script restart

Reply:
ok - I was thinking of the `#balances` property on the `BalancesPool`, which returns `{status: "initialising"}` and abstracts away the `#initialising` property of the pool - see https://github.com/TalismanSociety/talisman/blob/6d68f92adbd98bdea3699331d2742105ba545508/packages/extension-core/src/domains/balances/pool.ts#L124

Reply:
I think the semantics of `#balances` is:

- init background script
  - `status: "live"`
- create first balances subscription
  - `status: "initialising"`
     (at first due to `isRestartPending`, and then due to a non-0-length list of initialising balances)
- all balances initialised
  - `status: "live"`

Reply:
Correct, I tested it and confirmed! LGTM then